cmd/app/http.go

```diff
@@ -102,12 +106,32 @@ func createHTTPServer(ctx context.Context, serverEndpoint string, grpcServer, le
 	return httpServer{&api, serverEndpoint}
 }
 
-func (h httpServer) startListener() {
+func (h httpServer) startListener(wg *sync.WaitGroup) {
 	log.Logger.Infof("listening on http at %s", h.httpServerEndpoint)
+
+	idleConnsClosed := make(chan struct{})
+	go func() {
+		sigint := make(chan os.Signal, 1)
+		signal.Notify(sigint, syscall.SIGINT, syscall.SIGTERM)
+		<-sigint
+
+		// received an interrupt signal, shut down
+		if err := h.Shutdown(context.Background()); err != nil {
+			// error from closing listeners, or context timeout
+			log.Logger.Errorf("HTTP server Shutdown: %v", err)
+		}
+		close(idleConnsClosed)
+		log.Logger.Info("stopped http server")
+	}()
+
+	wg.Add(1)
 	go func() {
 		if err := h.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
```

Comment:
If the server fails to start immediately for some reason (e.g. the port is already in use), I think you'll be blocked waiting on `idleConnsClosed` to be written to but it won't be because the signal handler is still waiting (requiring intervention to stop). 

we probably should consider changing this to say if `err != nil && !errors.Is(err, http.ErrServerClosed)`, then `Fatal` log the error, otherwise just `Error` log the error and continue.

Reply:
Good call, I confirmed this hangs with just Error. We only need to log for fatal, err would only be for logging server closed.